### PR TITLE
Add examples for namespaceSelector via matchLabels and matchExpressions.

### LINKED ISSUE #1439: ClusterQueue namespace selector docs is not clear
**What would you like to be added**:

Example / Clarification on how to select on specific namespace either using `matchLabel` or `matchExpressions` for `ClusterQueue`.

**Why is this needed**:

From the docs its not clear how users can target a specific namespace in clusterqueue.

**Completion requirements**:

Examples of targeting a specific namespace or a list of namespace in the clusterqueue docs.

This enhancement requires the following artifacts:

- [ ] Design doc
- [ ] API change
- [x] Docs update

The artifacts should be linked in subsequent comments.

@alizaidis

### PR DESCRIPTION
**Pull request by alizaidis**
Fixes #1439

/kind documentation

### CODE REVIEW

**Review by alculquicondor — COMMENTED** [review]
Let's add a link to the full documentation for selectors: https://kubernetes.io/docs/reference/kubernetes-api/common-definitions/label-selector/

**Comment by moficodes** [comment]
@alizaidis could you resolve Aldo's comment?

**Comment by alculquicondor** [comment]
The link is actually already there, but maybe it's not prominent enough. Our assumption when writing this page was that users were familiar with labels and selectors, but it seems that it's not always the case.
Can you give it another pass with a fresh set of eyes and see how you can incorporate this other link https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/

**Comment by moficodes** [comment]
@alculquicondor as I understand it, the `namespaceSelector` in `ClusterQueue` not as generic as `labelSelector` is in K8s. It seems it only works on a single key which is `kubernetes.io/metadata.name` so I wonder if we need to make the label selector doc any more prominent.

**Comment by alculquicondor** [comment]
No, it works for any key that matches a label in the Namespace objects.

**Comment by moficodes** [comment]
That makes a lot more sense then.

**Comment by moficodes** [comment]
@alizaidis I think you are already adding that example?

**Comment by alizaidis** [comment]
> @alizaidis I think you are already adding that example?

Yes, commit incoming!

File: site/content/en/docs/concepts/cluster_queue.md [hunk 1]
```diff
@@ -129,15 +129,65 @@ in the `.spec.namespaceSelector` field.
...
+
+A namespaceSelector configuration allowing both namespaces to submit Jobs to this ClusterQueue would look something like this:
+
+```yaml
+namespaceSelector:
+  matchLabels:
+    research-cohort: team-a-b
+```
+
+Another way to configure `namespaceSelector` is using `matchExpressions` to match the workload to namespaces, for example a setup for namespaces `team-a` and `team-b` looks like the following:
```
Comment (alculquicondor) [inline site/content/en/docs/concepts/cluster_queue.md:170]: I think it's ok to stop here and refer to the Kubernetes documentation for how to use matchExpressions.

Comment (alizaidis) [reply site/content/en/docs/concepts/cluster_queue.md]: Added a link to Kubernetes documentation and removed the matchExpressions example from the page.

File: site/content/en/docs/concepts/cluster_queue.md [hunk 2]
```diff
@@ -129,17 +129,46 @@ in the `.spec.namespaceSelector` field.
...
+
+A namespaceSelector configuration allowing both namespaces to submit Jobs to this ClusterQueue would look something like this:
+
+```yaml
+namespaceSelector:
+  matchLabels:
+    research-cohort: team-a-b
+```
+
+Another way to configure `namespaceSelector` is using `matchExpressions` to match the workload to namespaces, see [Kubernetes documentation](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#resources-that-support-set-based-requirements) for more details.
```
Comment (alculquicondor) [inline site/content/en/docs/concepts/cluster_queue.md:170]:
```suggestion
Another way to configure `namespaceSelector` is using `matchExpressions`. See [Kubernetes documentation](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#resources-that-support-set-based-requirements) for more details.
```

**Comment by tenzen-y** [comment]
@alizaidis Can you address https://github.com/kubernetes-sigs/kueue/pull/1442#issuecomment-1858508593?
/release-note-none

**Comment by alizaidis** [comment]
@mimowo Rebase complete, would appreciate a review!

**Comment by mimowo** [comment]
/lgtm
Thanks for the update

**Comment by mimowo** [comment]
@alizaidis looking at https://github.com/kubernetes-sigs/kueue/pull/1442#issuecomment-1960410160 you should probably eliminate "Fixes #" from the commit message. This info goes to the PR description.

**Comment by tenzen-y** [comment]
@alizaidis Could you drop the message "Fixes https://github.com/kubernetes-sigs/kueue/issues/1439" from the commit message?

https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#github-keywords

**Comment by alculquicondor** [comment]
@moficodes can you rebase and fix the commit message?

**Comment by alizaidis** [comment]
Apologies for the delayed update, fixed the commit message!

**Pull request merged**